### PR TITLE
config/operator: Add operator YAMLs

### PR DESCRIPTION
**Pull request by kfirtoledo**
Add operator deployment(manager) and rbacs.
Note: All the config files were generated by the Kubebuildrer CLI.

References #286

### CODE REVIEW

File: config/operator/manager/manager.yaml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+kind: Namespace
+metadata:
+  labels:
+    control-plane: controller-manager
+    app.kubernetes.io/name: namespace
+    app.kubernetes.io/instance: clusterlink-operator-ns
+    app.kubernetes.io/component: manager
+    app.kubernetes.io/created-by: cl-operator
+    app.kubernetes.io/part-of: cl-operator
+  name: clusterlink-operator-ns
```
Comment (elevran) [inline config/operator/manager/manager.yaml:11]: is the choice of adding `-ns` Kubebuilder or input from the user?

File: config/operator/manager/manager.yaml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app.kubernetes.io/component: manager
```
Comment (elevran) [inline config/operator/manager/manager.yaml:17]: which of these (presence of key, the value assigned) are under our control, if any?

File: config/operator/manager/manager.yaml [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+      labels:
+        control-plane: controller-manager
+    spec:
+      containers:
+      - args:
+        - --secure-listen-address=0.0.0.0:8443
+        - --upstream=http://127.0.0.1:8080/
+        - --logtostderr=true
+        - --v=0
+        image: gcr.io/kubebuilder/kube-rbac-proxy:v0.15.0
```
Comment (elevran) [inline config/operator/manager/manager.yaml:43]: How is this container used?
What does the image do in the context of our work?

File: config/operator/rbac/auth_proxy_client_clusterrole.yaml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+apiVersion: rbac.authorization.k8s.io/v1
```
Comment (elevran) [inline config/operator/rbac/auth_proxy_client_clusterrole.yaml:1]: What is the `auth_porxy` and how relevant to our use case? 
Could it be something that kubebuilder creates for other use cases? For example for an Authorization hook or something else?

I think we need to understand what these YAMLs are for and only generate/include the ones needed by us.
IMHO, there are more objects than we really need.

File: config/operator/rbac/operator_editor_role.yaml [hunk 5]
```diff
@@ -0,0 +1,32 @@
+# permissions for end users to edit operators.
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
```
Comment (elevran) [inline config/operator/rbac/operator_editor_role.yaml:3]: the `clusterlink.net` objects are **namespaced**, thus I think we should have a Role and RoleBinding not cluster scoped roles and bindings.

File: config/operator/manager/manager.yaml [hunk 6]
```diff
@@ -0,0 +1,97 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  labels:
+    control-plane: controller-manager
+    app.kubernetes.io/name: namespace
+    app.kubernetes.io/instance: clusterlink-operator-ns
```
Comment (elevran) [inline config/operator/manager/manager.yaml:7]: same `operator` comment as on the other PR's

**Review by elevran — CHANGES_REQUESTED** [review]

File: config/operator/manager/manager.yaml (hunk 1 above)
Comment (kfirtoledo) [reply config/operator/manager/manager.yaml:11]: I added it, I can remove it...

File: config/operator/rbac/operator_editor_role.yaml (hunk 5 above)
Comment (kfirtoledo) [reply config/operator/rbac/operator_editor_role.yaml:3]: I will change it in the next phase, when we have the new controller

**Review by elevran — APPROVED** [review]

**Comment by kfirtoledo** [comment]
@elevran PTAL

File: config/operator/manager/manager.yaml [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  labels:
+    app.kubernetes.io/name: namespace
+    app.kubernetes.io/instance: clusterlink-system
+    app.kubernetes.io/created-by: cl-operator
+    app.kubernetes.io/part-of: cl-operator
+  name: clusterlink-system
```
Comment (elevran) [inline config/operator/manager/manager.yaml:82]: nit: GH complains of missing new line at EoF

File: config/operator/rbac/role_binding.yaml [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+  namespace: clusterlink-operator
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: cl-operator-manager-role
+subjects:
+- kind: ServiceAccount
+  name: cl-operator-controller-manager
+  namespace: clusterlink-operator
+
```
Comment (elevran) [inline config/operator/rbac/role_binding.yaml:21]: nit: EoL at EoF

**Review by elevran — APPROVED** [review]

**Pull request merged**